Workspace/src/arm_interface/src/arm_motor_commands_node.cpp

```diff
@@ -0,0 +1,108 @@
+#include <ros/ros.h>
+#include <std_msgs/String.h> //includes string library 
+#include <stdbool.h>
+#include <list>
+#include <vector>
+
+struct msg_from_topics {
+	bool ik_status; 
+	std::list<float> data_points; 
+};
+
+class messageReceivedFromGUIWrapper {
+
+	vector<msg_from_topics> msg;
+
+	void messageReceivedFromGUI(const msg_from_topics values) {
+		msg.add(values);
+	}
+	// To access certain index: msg.at(i); size: msg.size(); remove: msg.pop_back()
+}
+
+	
+
+
+
+// messageRecievedFromGUI return the message that is recieved from the Inverse Kinematics library after it passes the message parameter into the library
+// Not sure if this library exists or not yet
+msg_from_topics messageRecievedFromGUI(const msg_from_topics values) {
+	return values; 
+}
+
+// messageReceieveFromCAN has a message passed into it and returrns the message 
+std::list<float> messageRecievedFromCAN(const std::list<float> values) {
+	return values; 
+}
+
+// subscriberFunction is passed into the necessary parameters required to initiaize the node and a boolean that determines if the message that is recieved from the topic
+// needs to be passed into the Inverse Kinematics library or has it returned from it and needs to be returned as it is 
+msg_from_topics subscriberFunctionFromGUI (ros::NodeHandle nh, int argc, char** argv) {
+	//initializing the node
+	ros::init(argc, argv, "subscribe_arm_motor_commands");
+
+	ros::Subscriber sub = nh.subscribe("GUI", 10, messageReceivedFromGUI);
+
+	return ros::spin();		
+}
+
+std::list<float> subscriberFunctionFromCAN (ros::NodeHandle nh, int argc, char** argv) {
+	//initializing the node
+	ros::init(argc, argv, "subscribe_arm_motor_commands");
+
+	ros::Subscriber sub = nh.subscribe("CAN", 10, messageReceivedFromCAN);
+
+	return ros::spin();		
```

Comment:
Why is this here? this is a blocking call, it should be in your main function after all of your setup